File: cpp/83. 删除排序链表中的重复元素.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head == nullptr) {
            return head;
        }
        ListNode* slow = head;
        ListNode* fast = head;
        while(fast != nullptr){
            if(fast->val != slow->val){
                slow=slow->next;
                slow->val=fast->val;
            }
            fast=fast->next;
        }
        slow->next=nullptr;
        return head;
    }
};

File: 刷题/golang/83. 删除排序链表中的重复元素.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func deleteDuplicates(head *ListNode) *ListNode {
    if head == nil {
        return head
    }
    slow, fast := head, head
    for fast != nil {
        if slow.Val != fast.Val {
            slow=slow.Next
            slow.Val = fast.Val
        }
        fast=fast.Next
    }
    slow.Next = nil
    return head
}